src/App/App.test.js

```diff
@@ -25,25 +24,144 @@ describe("App", () => {
+    it("User is redirected to login page on link click", async () => {
+
+      render(
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>    
+      );
+
+      userEvent.click(screen.getByRole("link", {name: "Login"}));
+      expect(screen.getByText(/login/i)).toBeInTheDocument();
+      expect(screen.getByRole("button", {name: "Submit"})).toBeInTheDocument();
+    });
+    //when you put router inside of a component - makes it harder to test
+    //right now history is leaking from test above - starting at login page instead of home page
+    //quick fix: 
+    //best practice take out router in app component + make sure render app in test -> wrapped in <Router history={history}
+    it("Should route to a page with more movie details", async () => {
+
+      render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+      );
+      const movieTitle = await waitFor(() => screen.getByAltText("poster of Money Plane"));
+      expect(movieTitle).toBeInTheDocument()
+      userEvent.click(screen.getByRole('link', {name: /money plane/i}))
+      await waitFor(() => expect(screen.getByText("Runtime: 82 minutes")).toBeInTheDocument())
+    })
+    it("Should display custom dashboard and have option to rate a movie if a user is logged in", async () => {
+      postUserLogin.mockResolvedValue(
+        {
+          "user": {
+            "email": "marge@turing.io",
+            "id": 84,
+            "name": "Marge"
+          }
+        }
+      )
+
+      getUserRatings.mockResolvedValue(
+        { "ratings": 
+          [
+            {
+              "id": 2743,
+              "user_id": 84,
+              "movie_id": 694919,
+              "rating": 3,
+              "created_at": "2020-10-15T00:41:26.504Z",
+              "updated_at": "2020-10-15T00:41:26.504Z"
+            },
+            {
+              "id": 2747,
+              "user_id": 84,
+              "movie_id": 413518,
+              "rating": 4,
+              "created_at": "2020-10-15T01:28:13.417Z",
+              "updated_at": "2020-10-15T01:28:13.417Z"
+            },
+            {
+              "id": 2749,
+              "user_id": 84,
+              "movie_id": 550231,
+              "rating": 4,
+              "created_at": "2020-10-15T01:37:34.555Z",
+              "updated_at": "2020-10-15T01:37:34.555Z"
+            }
+          ]
+        }
+      )
+
+      render (
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>
+       )
+       userEvent.click(screen.getByRole("link", {name: "Login"}));
+       userEvent.type(screen.getByPlaceholderText("email"), "marge@turing.io");
+       userEvent.type(screen.getByPlaceholderText("password"), "password123");
+       userEvent.click(screen.getByRole("button", {name: "Submit"}));
+       await waitFor(() => expect(screen.getByText("Logout")).toBeInTheDocument())
+       expect(screen.getByText("Your rating: 3")).toBeInTheDocument()
+    })
+
   });
+})
+
+
```

Comment:
I think there is still a few more integration / unit tests we can add to me our suite a bit more robust.